gremlin-scala/src/main/scala/gremlin/scala/ScalaEdge.scala

```diff
@@ -29,9 +30,15 @@ case class ScalaEdge(edge: Edge) extends ScalaElement[Edge] {
   }
 
   def toCC[T <: Product: Marshallable] =
-    implicitly[Marshallable[T]].toCC(edge.id, edge.valueMap())
+    implicitly[Marshallable[T]].toCC(edge.id, edge.valueMap)
 
-  def start() = GremlinScala[Edge, HNil](__.__(edge))
+  override def start() = GremlinScala[Edge, HNil](__.__(edge))
+
+  override def properties[A]: Stream[Property[A]] =
+    edge.properties[A](keys.toSeq: _*).toStream
+
+  override def properties[A](wantedKeys: String*): Stream[Property[A]] =
```

Comment:
for consistency, can we change `wantedKeys` to `keys`? it's often useful to call a function with named parameters, so it would also be better for brevity on the call side